projects/core/src/user/facade/user.service.ts

```diff
@@ -27,14 +28,28 @@ export class UserService {
   constructor(
     protected store: Store<
       fromStore.StateWithUser | fromProcessStore.StateWithProcess<void>
-    >
+    >,
+    protected authStore: Store<fromAuthStore.StateWithAuth>
   ) {}
 
   /**
    * Returns a user
    */
   get(): Observable<User> {
-    return this.store.pipe(select(fromStore.getDetails));
+    return this.store.pipe(select(fromAuthStore.getUserToken)).pipe(
```

Comment:
I think, there is no need to select the `userToken`. Just `getUserDetails`, if user data is empty, then call `load`. If there is no `user token`, this `load` will be failed. Error message will be displayed. 

If you check other functions in `user.service.ts` or `checkout.service.ts`, most `getXXXX` or `loadXXXX` functions require user login, but none of them select 'userToken`, because for anonymous users, these functions should not be called at all. If they are called, then we should see the error.

Reply:
Fixed in #2865